sklearn/linear_model/_logistic.py

```diff
@@ -18,6 +18,12 @@
 from scipy.special import expit, logsumexp
 from joblib import Parallel, delayed, effective_n_jobs
 
+from ..utils._typing import Literal
+from ..utils._typing import RandomState
+from ..utils._typing import Annotated
+from ..utils._typing import Shape
+from typing import Union
+from typing import Optional
```

Comment:
Any reason why these are all on a separate line?

Reply:
It's a personal style choice so diffs are easier to look at. I also like:

```py
from typing import (
	Union,
	Optional,
)
```

Reply:
True as well.

Unrelated: Does sklearn uses [black](https://pypi.org/project/black/)?

Reply:
We are slowly working our way on deciding this: https://github.com/scikit-learn/scikit-learn/issues/11336

Reply:
I think this comment is relevant here: https://github.com/scikit-learn/scikit-learn/issues/11336#issuecomment-492482002

If we go with adding typing information, then black is reasonable for its style choices.